### ISSUE TITLE
discuss: `hmac-auth` use Http Date to do time offset check

### ISSUE THREAD

**Issue by gxthrj** [apache/apisix#2287]
### Issue description
The unit of key `X-HMAC-TIMESTAMP` in plugin `hmac-auth` is an optional signature string,
Time offset verification through http `Date` and `clock_skew`

http `Date` use GMT format, and to be a part of sign string.

ref:
https://www.alibabacloud.com/help/doc-detail/25492.htm
https://docs.aws.amazon.com/AmazonS3/latest/dev/RESTAuthentication.html

### Environment

* apisix version: 1.5
* OS:

**Comment by moonming** [comment]
please add link for iso 8601

**Comment by moonming** [comment]
please show a example

**Comment by moonming** [comment]
I don't think `http Date` is a good name, should be `date` or `x-date`

**Comment by nic-chen** [comment]
> I don't think `http Date` is a good name, should be `date` or `x-date`

yes, it's `Date` not `http Date`

**Comment by gxthrj** [comment]
yes , http `Date` ,the `D` is uppercase

**Comment by gxthrj** [comment]
> please show a example

 a example in aws  https://docs.aws.amazon.com/AmazonS3/latest/dev/RESTAuthentication.html#RESTAuthenticationExamples

**Comment by moonming** [comment]
> yes , http `Date` ,the `D` is uppercase

No, http header is case-insensitive. ref https://tools.ietf.org/html/rfc7230#section-3.2:
```
Each header field consists of a case-insensitive field name followed
   by a colon (":"), optional leading whitespace, the field value, and
   optional trailing whitespace.
```

**Comment by gxthrj** [comment]
> > yes , http `Date` ,the `D` is uppercase
> 
> No, http header is case-insensitive. ref https://tools.ietf.org/html/rfc7230#section-3.2:
> 
> ```
> Each header field consists of a case-insensitive field name followed
>    by a colon (":"), optional leading whitespace, the field value, and
>    optional trailing whitespace.
> ```

I am not sure if the `Date` is the same as other headers
ref: https://tools.ietf.org/html/rfc7231#section-7.1.1.2

**Comment by nic-chen** [comment]
@gxthrj 
please confirm using `GMT` format not `ISO 8601`, right ?

**Comment by gxthrj** [comment]
> @gxthrj
> please confirm using `GMT` format not `ISO 8601`, right ?

yes, http `Date` is GMT, not iso 8601

**Comment by gxthrj** [comment]
it is merged with #2301
